File: OOP C++/Templete/Generic Class/Demo1.cpp
#include <iostream>
using namespace std;
template <class A>
class Demo1
{
    A x, y, z;

public:
    Demo1(A a, A b, A c)
    {
        x = a;
        y = b;
        z = c;
    }
    void show()
    {
        cout << "Sum of x,y & z is : " << x + y + z << endl;
    }
    void show2()
    {
        cout << x << "\t" << y << "\t" << z << endl;
    }
    A display()
    {
        return x * y * z;
    }
};
int main()
{
    Demo1<int> ob(2, 3, 4);
    Demo1<char> ob2('I', 'H', 'S');
    Demo1<string> ob3("Imam", "Hasan", "Shawon");
    Demo1<double> ob4(2.2, 3.3, 4.5);
    cout << "\n";

    ob.show();
    cout << "The multiplication is : " << ob.display() << endl;
    ob2.show2();
    ob3.show2();

    ob4.show();
    cout << "The multiplication is : " << ob4.display() << endl;

    getchar();
    return 0;
}

File: OOP Java/Variable_Length_args/Demo1.java
package Variable_Length_args;

public class Demo1 {

    void display(int... n) {
        int s=0;
        for (int x : n) {
            s += x;
        }
        System.out.println("Sum : " + s);
    }
    
    public static void main(String[] args) {
        Demo1 ob = new Demo1();
        ob.display(1, 2, 3);
        ob.display(44, 88);
        ob.display(1, 5, 6, 9);
    }
}

